samples/basic/rgb_led/src/main.c

```diff
@@ -5,103 +5,130 @@
-    DT_PHA_HAS_CELL(DT_ALIAS(green_pwm_led), pwms, channel) && \
-    DT_NODE_HAS_PROP(DT_ALIAS(blue_pwm_led), pwms) && \
-    DT_PHA_HAS_CELL(DT_ALIAS(blue_pwm_led), pwms, channel)
-/* Get the defines from dt (based on aliases 'red-pwm-led', 'green-pwm-led' &
- * 'blue-pwm-led')
+/*
+ * Devicetree helper macro which gets the 'flags' cell from a 'pwms'
+ * property, or returns 0 if the property has no 'flags' cell.
  */
-#define PWM_DEV0	DT_PWMS_LABEL(DT_ALIAS(red_pwm_led))
-#define PWM_CH0		DT_PWMS_CHANNEL(DT_ALIAS(red_pwm_led))
-#if DT_PHA_HAS_CELL(DT_ALIAS(red_pwm_led), pwms, flags)
-#define PWM_FLAGS0	DT_PWMS_FLAGS(DT_ALIAS(red_pwm_led))
-#else
-#define PWM_FLAGS0	0
-#endif
-#define PWM_DEV1	DT_PWMS_LABEL(DT_ALIAS(green_pwm_led))
-#define PWM_CH1		DT_PWMS_CHANNEL(DT_ALIAS(green_pwm_led))
-#if DT_PHA_HAS_CELL(DT_ALIAS(green_pwm_led), pwms, flags)
-#define PWM_FLAGS1	DT_PWMS_FLAGS(DT_ALIAS(green_pwm_led))
+
+#define FLAGS_OR_ZERO(node)						\
+	COND_CODE_1(DT_PHA_HAS_CELL(node, pwms, flags),			\
+		    (DT_PWMS_FLAGS(node)),				\
+		    (0))
+
+/*
+ * Extract devicetree configuration.
+ */
+
+#define RED_NODE DT_ALIAS(red_pwm_led)
+#define GREEN_NODE DT_ALIAS(green_pwm_led)
+#define BLUE_NODE DT_ALIAS(blue_pwm_led)
+
+#if DT_HAS_NODE_STATUS_OKAY(RED_NODE)
+#define RED_LABEL	DT_PWMS_LABEL(RED_NODE)
+#define RED_CHANNEL	DT_PWMS_CHANNEL(RED_NODE)
+#define RED_FLAGS	FLAGS_OR_ZERO(RED_NODE)
 #else
-#define PWM_FLAGS1	0
+#error "Unsupported board: red-pwm-led devicetree alias is not defined"
+#define RED_LABEL	""
+#define RED_CHANNEL	0
+#define RED_FLAGS	0
 #endif
-#define PWM_DEV2	DT_PWMS_LABEL(DT_ALIAS(blue_pwm_led))
-#define PWM_CH2		DT_PWMS_CHANNEL(DT_ALIAS(blue_pwm_led))
-#if DT_PHA_HAS_CELL(DT_ALIAS(blue_pwm_led), pwms, flags)
-#define PWM_FLAGS2	DT_PWMS_FLAGS(DT_ALIAS(blue_pwm_led))
+
+#if DT_HAS_NODE_STATUS_OKAY(GREEN_NODE)
+#define GREEN_LABEL	DT_PWMS_LABEL(GREEN_NODE)
+#define GREEN_CHANNEL	DT_PWMS_CHANNEL(GREEN_NODE)
+#define GREEN_FLAGS	FLAGS_OR_ZERO(GREEN_NODE)
 #else
-#define PWM_FLAGS2	0
+#error "Unsupported board: green-pwm-led devicetree alias is not defined"
+#define GREEN_LABEL	""
+#define GREEN_CHANNEL	0
+#define GREEN_FLAGS	0
 #endif
+
+#if DT_HAS_NODE_STATUS_OKAY(BLUE_NODE)
+#define BLUE_LABEL	DT_PWMS_LABEL(BLUE_NODE)
+#define BLUE_CHANNEL	DT_PWMS_CHANNEL(BLUE_NODE)
+#define BLUE_FLAGS	FLAGS_OR_ZERO(BLUE_NODE)
 #else
-#error "Choose supported board or add new board for the application"
+#error "Unsupported board: blue-pwm-led devicetree alias is not defined"
+#define BLUE_LABEL	""
+#define BLUE_CHANNEL	0
+#define BLUE_FLAGS	0
 #endif
 
 /*
  * 50 is flicker fusion threshold. Modulated light will be perceived
  * as steady by our eyes when blinking rate is at least 50.
  */
-#define PERIOD (USEC_PER_SEC / 50U)
-
-/* in micro second */
-#define STEPSIZE	2000
+#define PERIOD_USEC	(USEC_PER_SEC / 50U)
+#define STEPSIZE_USEC	2000
 
 static int write_pin(struct device *pwm_dev, u32_t pwm_pin,
```

Comment:
Would you mind renaming this function to something more sensible? Maybe `pwm_start`?

Reply:
Will do. I'm going to rebase on #25214 first thing tomorrow, will do that then.

Reply:
> Will do. I'm going to rebase on #25214 first thing tomorrow, will do that then.

Now renamed to "pwm_set"